src/lib/proxie.ts

```diff
@@ -0,0 +1,339 @@
+const inquirer = require('inquirer');
+const fuzzy = require('fuzzy');
+import getPort from 'get-port';
+import { Answers } from 'inquirer';
+import CloudSqlProxyCli from './CloudSqlProxyCli';
+import GcloudCli, { Cluster, Project, SqlInstance, VMInstance } from './GcloudCli';
+import KubectlCli, { Pod } from './KubectlCli';
+const commandExists = require('command-exists');
+
+export const globalOptions = {
+    debug: !!/bescripts/.test(String(process.env.DEBUG)),
+};
+
+const gcloud = new GcloudCli(globalOptions);
+const kubectl = new KubectlCli(globalOptions);
+const cloudSqlProxy = new CloudSqlProxyCli(globalOptions);
+
+inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
+
+const initializationCheck = async () => {
+    console.log('Command check:');
+    const missing = (await Promise.all(['gcloud', 'kubectl', 'cloud_sql_proxy']
+        .map(async cmd => {
+            let status = '❌';
+            try {
+                await commandExists(cmd);
+                status = '√';
+            } catch (error) {}
+            console.log(` ${cmd} ${status}`);
+            return { status, cmd };
+        })
+    ))
+        .filter(({ status }) => status !== '√');
+    if (missing.length) {
+        console.log(`Missing following commands: ${missing.map(x => x.cmd)}. Will now exit.`);
+        process.exit(1);
+    }
+    const auth = await gcloud.configListAccount();
+    console.log(`Active gcloud account: ${auth && auth.core && auth.core.account}`);
+};
+
+const vmPicker = () => {
+    let list = Promise.resolve([]) as Promise<Array<VMInstance>>;
+    const getList = async (answers: any) => {
+        const data = await list;
+        if (data.length) {
+            return data;
+        }
+        list = gcloud.computeInstancesList({ project: answers.project });
+        return list;
+    };
+    return {
+        type: 'autocomplete',
+        name: 'vminstance',
+        message: 'VM Instance (mongo...)',
```

Comment:
Why `mongo`?

Reply:
I dont get the question.. why not?

The hint should be for developers to proxy to a service. And in our cases, mongo instances are either in pods, or in an standalone vm instance.

I will be glad to change this to anything that makes more sense.